### DISCUSSION TITLE
Howto debug fluxcd kustomization patch errors.

### DISCUSSION THREAD

**Discussion by jwausle** [fluxcd/flux2#2322] [Q&A]
I observe a warning in my installed Kustomization with a patch. 

```
Warning  error   2m13s  kustomize-controller  Kustomization/default/my-kustomization-with-patch dry-run failed, error: fa │
│ iled to create typed patch object: .spec.name: field not declared in schema
```

For me it's now unclear what can I do. I need more informations about the warning. 

Hint: The same patchesStrategicMerge with  `kustomize build` works well.

Questions:

1. Is it possible to debug the flux patch
2. Is it possible to simulate the kustomization locally - e.g. `flux build kustomization` like `kustomize` can

**Comment by kingdonb** [comment]
It will be difficult to diagnose your issue without having provided the manifests that are causing an error. Can you post them?

https://fluxcd.io/blog/2022/01/january-update/#-new-features-and-improvements - the January Flux update included a new feature called `flux diff` which might help with debugging. This will simulate the kustomization locally and tell you what Flux sees as the update to apply, even before it has been committed and pushed. (Hopefully this new feature is as game-changing as it looks!)

**Reply by 4c74356b41** [reply]
no matter what I do with `flux diff` i get:
```
not a valid directory: fs-security-constraint abs C:\Users\xxx\yyy\zzz\infrastructure: path 'C:\Users\xxx\yyy\zzz\infrastructure' is not in or below 'C:\'
```

is there an actual way to debug the kustomization process inside flux? i can see my kustomization file in the cluster, but patches are either not processed at all (i see nothing in `flux logs`) or somehow failing
